### PR TITLE
added support for NIP-62 Request to Vanish events #80

### PR DESCRIPTION
**Pull request by bryanmontz**
## Issues covered
#80 

## Description
Adds support for authoring NIP-62 Request to Vanish events.

Note: This is only the data layer. The UI changes required for the user to make this request will be added in a subsequent PR.

No UI changes.

Initially I added a function for the new event to CurrentUser.swift, but that triggered the 500-line file lint rule. So I made an extension on CurrentUser for all the publish event functions.

This PR also adds the concept of an extension on JSONEvent that could house easier-to-use APIs for event creation. It also improves testability.

### CODE REVIEW

File: README.md
Comment (joshuatbrown) [inline README.md:1]: Thanks for updating the README!

File: Nos/Service/CurrentUser.swift
Comment (joshuatbrown) [inline Nos/Service/CurrentUser.swift:1]: Thanks for refactoring this -- much needed.

**Review by joshuatbrown — COMMENTED** [review]
This all looks great! A couple questions:

If we see a request to vanish for a user, should we delete all events for that user? And if so, should that be in this PR or a separate one?

**Comment by bryanmontz** [comment]
> This all looks great! A couple questions:
> 
> If we see a request to vanish for a user, should we delete all events for that user? And if so, should that be in this PR or a separate one?

I would think that would be in a separate PR, but yeah, we should do that.

**Comment by bryanmontz** [comment]
@joshuatbrown Based on the title of this PR, you're probably right to assume that we should make your suggested change here. I still think it would be better to separate that though. Here's a ticket for it: https://github.com/planetary-social/nos/issues/1511.

**Review by joshuatbrown — APPROVED** [review]
Excellent work again, @bryanmontz!

**Pull request merged**